src/main/java/com/dropbox/core/android/AuthActivity.java

```diff
@@ -529,10 +617,35 @@ protected void onNewIntent(Intent intent) {
             }
 
             // Successful auth.
-            newResult = new Intent();
-            newResult.putExtra(EXTRA_ACCESS_TOKEN, token);
-            newResult.putExtra(EXTRA_ACCESS_SECRET, secret);
-            newResult.putExtra(EXTRA_UID, uid);
+            if (token.equals(TokenType.OAUTH2.toString())) {
+                // token flow
+                newResult = new Intent();
+                newResult.putExtra(EXTRA_ACCESS_TOKEN, token);
+                newResult.putExtra(EXTRA_ACCESS_SECRET, secret);
+                newResult.putExtra(EXTRA_UID, uid);
+            } else if (token.equals(TokenType.OAUTH2CODE.toString())) {
+                // code flow with PKCE
+                TokenRequest tokenRequest = new TokenRequest(secret);
+                try {
+                    DbxAuthFinish dbxAuthFinish = tokenRequest.execute().get();
```

Comment:
The preferred solution is actually to use a raw Thread since this is a network call.

If you prefer to use an AsyncTask, instead of `execute()` use `executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR)` otherwise this request might block or be blocked by other tasks.